### PR TITLE
fix(link-button): updates default style for link buttons

### PR DESCRIPTION
**Pull request by zahnster**
* Updates default size prop for Link buttons to 'default' from 'reset', to ensure proper font & line height sizing.

### CODE REVIEW

File: packages/paste-core/components/button/src/index.tsx [hunk 1]
```diff
@@ -30,14 +30,14 @@ import {ResetButton} from './ResetButton';
...
-// - 'reset' for variant 'link'
+// - 'reset' for reset
 // - 'icon' if there's 1 child that's an icon
 // - 'default' otherwise
 const getButtonSize = (variant: ButtonVariants, children: React.ReactNode, size?: ButtonSizes): ButtonSizes => {
   let smartSize: ButtonSizes = 'default';
   if (size != null) {
     smartSize = size;
-  } else if (variant === 'link' || variant === 'destructive_link' || variant === 'reset') {
+  } else if (variant === 'reset') {
```
Comment (SiTaggart) [inline packages/paste-core/components/button/src/index.tsx:40]: So the reason you can't use the default size for links is that default sizing comes with padding. 

image

As I mentioned in Slack, the real fix is to make sure reset, at the very least _sets a font size and line height_ by explicitly inheriting them from their surrounding text.

Which I suspect is exactly the behaviour you would get from an Anchor component too. It inherits the parent font and line height sizes

**Review by SiTaggart — CHANGES_REQUESTED** [review]

File: packages/paste-core/components/button/src/index.tsx (hunk 1 above)
Comment (zahnster) [reply packages/paste-core/components/button/src/index.tsx]: I see, I see. I did see an inherit font size property, but that was exclusively on the ResetButton component, and not in the broader base styles for the size prop. Adjusted the code, hopefully this is closer to what you had in mind.

File: .changeset/lucky-lies-travel.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@twilio-paste/button': patch
+'@twilio-paste/core': patch
+---
+
+Updates default size for link buttons to 'default' from 'reset'
```
Comment (andipants12) [inline .changeset/lucky-lies-travel.md:6]: Could you please update this to reflect the change we pivoted to?

**Review by andipants12 — COMMENTED** [review]
Looks great, just need to update the changeset message to reflect the direction we decided to go for a fix. 💯

File: .changeset/lucky-lies-travel.md (hunk 2 above)
Comment (zahnster) [reply .changeset/lucky-lies-travel.md:6]: _great_ point, yes

**Review by SiTaggart — APPROVED** [review]
Note worth mentioning. Our button stories are not super useful for link variant. 

I don't think we really intend on using link style button with actual sizes, so the "it's just a link button" case isn't very well captured in the VRT. But it does what it should do

**Comment by zahnster** [comment]
> Note worth mentioning. Our button stories are not super useful for link variant.
> 
> I don't think we really intend on using link style button with actual sizes, so the "it's just a link button" case isn't very well captured in the VRT. But it does what it should do

I noticed this 😅 . Locally, I verified through the website, as the docs examples are a bit more authentic to the expected implementations. I have _thoughts_ about our Button stories, but that's for another day...

**Review by andipants12 — APPROVED** [review]
:shipit:

**Pull request merged**